src/ompl/geometric/planners/rrt/STRRTstar.h

```diff
@@ -0,0 +1,320 @@
+/*********************************************************************
+* Software License Agreement (BSD License)
+*
+*  Copyright (c) 2021, Technische Universität Berlin (TU Berlin)
+*  All rights reserved.
+*
+*  Redistribution and use in source and binary forms, with or without
+*  modification, are permitted provided that the following conditions
+*  are met:
+*
+*   * Redistributions of source code must retain the above copyright
+*     notice, this list of conditions and the following disclaimer.
+*   * Redistributions in binary form must reproduce the above
+*     copyright notice, this list of conditions and the following
+*     disclaimer in the documentation and/or other materials provided
+*     with the distribution.
+*   * Neither the name of the TU Berlin nor the names of its
+*     contributors may be used to endorse or promote products derived
+*     from this software without specific prior written permission.
+*
+*  THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS
+*  "AS IS" AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT
+*  LIMITED TO, THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS
+*  FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT SHALL THE
+*  COPYRIGHT OWNER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+*  INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING,
+*  BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+*  LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+*  CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT
+*  LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+*  ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+*  POSSIBILITY OF SUCH DAMAGE.
+*********************************************************************/
+
+/* Author: Francesco Grothe */
+
+#ifndef OMPL_GEOMETRIC_PLANNERS_RRT_STRRT_STAR_
+#define OMPL_GEOMETRIC_PLANNERS_RRT_STRRT_STAR_
+
+#include <ompl/datastructures/NearestNeighbors.h>
+#include <ompl/base/goals/GoalSampleableRegion.h>
+#include <ompl/geometric/planners/PlannerIncludes.h>
+#include <ompl/base/OptimizationObjective.h>
+#include <ompl/base/spaces/SpaceTimeStateSpace.h>
+#include <ompl/base/objectives/MinimizeArrivalTime.h>
+#include <ompl/base/samplers/ConditionalStateSampler.h>
+#include <ompl/tools/config/SelfConfig.h>
+#include <ompl/util/GeometricEquations.h>
+#include <boost/math/constants/constants.hpp>
+
+namespace ompl
+{
+    namespace geometric
+    {
+        class STRRTstar : public base::Planner
```

Comment:
Could you add a small description of the algorithm here, similar to e.g. RRTConnect?